### PR TITLE
[docs] Add instructions for Docker image verification

### PR DESCRIPTION
**Pull request by kilfoyle**
This updates the legacy [Run APM Server on Docker](https://www.elastic.co/guide/en/apm/guide/current/running-on-docker.html) page with a step to validate the Docker image. I'm not sure if we update these legacy docs anymore, but based on the issue linked below I think this is an important addition.

Rel: https://github.com/elastic/dev/issues/2002

Preview

--- 
Screenshot 2023-05-23 at 2 51 37 PM

### CODE REVIEW

File: docs/legacy/copied-from-beats/docs/shared-docker.asciidoc [hunk 1]
```diff
@@ -29,11 +29,23 @@ endif::[]
...
 ------------------------------------------------
 docker pull {dockerimage}
 ------------------------------------------------
 
+. Verify the Docker image:
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+wget https://artifacts.elastic.co/cosign.pub
+cosign verify --key cosign.pub {docker-repo}:{version}
```
Comment (mgreau) [inline docs/legacy/copied-from-beats/docs/shared-docker.asciidoc:44]:
```suggestion
cosign verify --key cosign.pub {dockerimage}:{version}
```

Comment (kilfoyle) [reply docs/legacy/copied-from-beats/docs/shared-docker.asciidoc:44]: nice catch!

**Review by mgreau — APPROVED** [review]
LGTM

**Comment by dedemorton** [comment]
I'm going to tag in @bmorelli25 because he's more familiar with these docs.

**Review by bmorelli25 — APPROVED** [review]
> I'm not sure if we update these legacy docs anymore, but based on the issue linked below I think this is an important addition.

We do! Thank you for the update.

**Comment by bmorelli25** [comment]
Force pushed to sign commits.

**Review by simitt — APPROVED** [review]

**Comment by kilfoyle** [comment]
@dedemorton  or @bmorelli25 would you mind merging this for me since I'm not authorized?
Thanks!

**Pull request merged**